lib/services/role_test.go

```diff
@@ -3302,8 +3304,163 @@ func TestApplyTraits(t *testing.T) {
 				},
 			},
 		},
-	}
+		{
+			comment: "kubernetes namespaces preservation",
+			inTraits: map[string][]string{
+				"users": {"alice", "bob"},
+			},
+			allow: rule{
+				inKubeResources: []types.KubernetesResource{
+					{
+						Namespace: "default",
+						Kind:      "*",
+						Name:      "name-*",
+						Verbs:     []string{"get", "list"},
+					},
+				},
+				outKubeResources: []types.KubernetesResource{
+					{
+						Namespace: "default",
+						Kind:      "*",
```

Comment:
nit: the test values for `types.KubernetesResource` is almost same for every case, we might use a function to avoid repetition here. 
```
Kind:      "*",
Name:      "name-*",
Verbs:     []string{"get", "list"},
```